ruma-events/src/room/message.rs

```diff
@@ -85,6 +85,337 @@ impl MessageEventContent {
+            relates_to: Some(Relation::Reply {
+                in_reply_to: InReplyTo { event_id: original_message.event_id.clone() },
+            }),
+            #[cfg(feature = "unstable-pre-spec")]
+            new_content: None,
+        }
+    }
+
+    fn get_plain_quote_fallback(original_message: &MessageEvent) -> String {
+        match &original_message.content.msgtype {
+            MessageType::Audio(_) => {
+                format!("> <{:?}> sent an audio file.", original_message.sender)
+            }
+            MessageType::Emote(content) => {
+                format!("> * <{:?}> {}", original_message.sender, content.body)
+            }
+            MessageType::File(_) => {
+                format!("> <{:?}> sent a file.", original_message.sender)
+            }
+            MessageType::Image(_) => {
+                format!("> <{:?}> sent an image.", original_message.sender)
+            }
+            MessageType::Location(content) => {
+                format!("> <{:?}> {}", original_message.sender, content.body)
+            }
+            MessageType::Notice(content) => {
+                format!("> <{:?}> {}", original_message.sender, content.body)
+            }
+            MessageType::ServerNotice(content) => {
+                format!("> <{:?}> {}", original_message.sender, content.body)
+            }
+            MessageType::Text(content) => {
+                format!("> <{:?}> {}", original_message.sender, content.body)
+            }
+            MessageType::Video(_) => {
+                format!("> <{:?}> sent a video.", original_message.sender)
+            }
+            MessageType::_Custom(content) => {
+                // Eagerly deserialize `body` since we assume every message should have it
+                format!(
+                    "> <{:?}> {}",
+                    original_message.sender,
+                    content.data["body"].as_str().unwrap_or("")
+                )
+            }
+            #[cfg(feature = "unstable-pre-spec")]
+            MessageType::VerificationRequest(content) => {
+                // Eagerly deserialize `body` since we assume every message should have it
+                format!("> <{:?}> {}", original_message.sender, content.body)
+            }
+        }
+    }
+
+    fn get_html_quote_fallback(original_message: &MessageEvent) -> String {
+        match &original_message.content.msgtype {
+            MessageType::Audio(_) => {
+                format!(
+                    "<mx-reply>
+    <blockquote>
+        <a href=\"https://matrix.to/#/!{}/${}\">In reply to</a>
+        <a href=\"https://matrix.to/#/{}\">{}</a>
+        <br />
+        sent an audio file.
+    </blockquote>
+</mx-reply>",
+                    original_message.room_id.server_name(),
+                    original_message.event_id.as_str(),
+                    original_message.sender.as_str(),
+                    original_message.sender.as_str(),
+                )
```

Comment:
You can the same `format!` argument multiple times by explicitly using the index, i.e.

```rust
        <a href=\"https://matrix.to/#/!{0}/${1}\">In reply to</a>
        <a href=\"https://matrix.to/#/{2}\">{2}</a>
```

or by simply naming one or all of the parameters:

```suggestion
                format!(
                    "<mx-reply>
    <blockquote>
        <a href=\"https://matrix.to/#/{room_id}/{event_id}\">In reply to</a>
        <a href=\"https://matrix.to/#/{sender}\">{sender}</a>
        <br />
        sent an audio file.
    </blockquote>
</mx-reply>",
                    room_id = original_message.room_id,
                    event_id = original_message.event_id,
                    sender = original_message.sender,
                )
```

You may also notice I removed the extra `!` and `$` from the format string, which would have been duplicated, as well as removing

Reply:
Also could you look into replacing the `format!` with `formatdoc!` ([from indoc](https://docs.rs/indoc/1.0.3/indoc/macro.formatdoc.html)) so the indentation doesn't look so weird?

Reply:
Replaced all the `format!` calls with`formatdoc!` ones.
Also changed every call to use named arguments for readability.